validation/rbac/deprecated_restrictedadmin_test.go

```diff
@@ -0,0 +1,169 @@
+			require.NoError(rb.T(), err)
+		} else {
+			err := users.AddClusterRoleToUser(rb.client, rb.cluster, user, role.String(), nil)
+			require.NoError(rb.T(), err)
+		}
+	}
+
+	standardClient, err = standardClient.ReLogin()
+	require.NoError(rb.T(), err)
+
+	additionalUser, err := users.CreateUserWithRole(rb.client, users.UserConfig(), rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	rb.Run("Validating Global Role Binding is created for "+role.String(), func() {
+		verifyRAGlobalRoleBindingsForUser(rb.T(), user, rb.client)
+	})
+	rb.Run("Validating corresponding role bindings for users", func() {
+		verifyRARoleBindingsForUser(rb.T(), user, rb.client, rb.cluster.ID)
+	})
+	rb.Run("Validating if "+role.String()+" can list any downstream clusters", func() {
+		verifyRAUserCanListCluster(rb.T(), rb.client, standardClient, rb.cluster.ID, role)
+	})
+	rb.Run("Validating if members with role "+role.String()+" are able to list all projects", func() {
+		verifyRAUserCanListProject(rb.T(), rb.client, standardClient, rb.cluster.ID)
+	})
+	rb.Run("Validating if members with role "+role.String()+" is able to create a project in the cluster", func() {
+		verifyRAUserCanCreateProjects(rb.T(), standardClient, rb.cluster.ID, role)
+	})
+	rb.Run("Validate namespaces checks for members with role "+role.String(), func() {
+		verifyRAUserCanCreateNamespace(rb.T(), standardClient, adminProject, role)
+	})
+	rb.Run("Validating if "+role.String()+" can lists all namespaces in a cluster.", func() {
+		verifyRAUserCanListNamespace(rb.T(), rb.client, standardClient, rb.cluster.ID, role)
+	})
+	rb.Run("Validating if "+role.String()+" can delete a namespace from a project they own.", func() {
+		verifyRAUserCanDeleteNamespace(rb.T(), rb.client, standardClient, adminProject, rb.cluster.ID, role)
+	})
+	rb.Run("Validating if member with role "+role.String()+" can add members to the cluster", func() {
+		verifyRAUserCanAddClusterRoles(rb.T(), rb.client, standardClient, rb.cluster)
+	})
+	rb.Run("Validating if member with role "+role.String()+" can add members to the project", func() {
+		if strings.Contains(role.String(), "project") {
+			verifyRAUserCanAddProjectRoles(rb.T(), standardClient, adminProject, additionalUser, rbac.ProjectOwner.String(), rb.cluster.ID)
+		}
+	})
+	rb.Run("Validating if member with role "+role.String()+" can delete a project they are not owner of ", func() {
+		rbac.VerifyUserCanDeleteProject(rb.T(), standardClient, adminProject, role)
+	})
+	rb.Run("Validating if member with role "+role.String()+" is removed from the cluster and returns nil clusters", func() {
+		if strings.Contains(role.String(), "cluster") {
+			verifyRAUserCanRemoveClusterRoles(rb.T(), rb.client, user)
+		}
+	})
+}
+
+func (rb *RBTestSuite) TestRBAC() {
+	tests := []struct {
+		name   string
+		role   rbac.Role
+		member string
+	}{
+		{"Restricted Admin", restrictedAdmin, restrictedAdmin.String()},
+	}
+
+	for _, tt := range tests {
```

Comment:
nit, can we remove the for loop here as this test is for a single role.

Reply:
Even though its for demonstration, it will still be merged and used in recurring runs. Since the test will be part of the recurring runs, I think it makes sense to modify it in this PR itself.